src/Contracts/MonitoredScheduledTask.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace Spatie\ScheduleMonitor\Contracts;
+
+use Illuminate\Console\Events\ScheduledTaskFinished;
+use Illuminate\Console\Events\ScheduledTaskSkipped;
+use Illuminate\Console\Events\ScheduledTaskStarting;
+use Illuminate\Console\Scheduling\Event;
+use Illuminate\Database\Eloquent\Relations\HasMany;
+use OhDear\PhpSdk\Resources\CronCheck;
+use Spatie\ScheduleMonitor\Models\MonitoredScheduledTaskLogItem;
+
+interface MonitoredScheduledTask
```

Comment:
Instead of interface, I would like it better if, everywhere we need the models, we would check if the model given is or extends the one that ships with the package. 

This way people could still override all functions by providing a custom model.  And as a package maintainer we would be able to add extra functions in a non-breaking way.

Reply:
@freekmurze No problem. The thought here was when I was using it locally, we could just extend the existing models to override anything needed but at least they would still be constrained to the contract :) 

For example, in my use case I simply did this to update the database connection.
```php
class MonitoredScheduledTask extends ModelsMonitoredScheduledTask
{
    protected $connection = 'log';
}
```

Let me know what you think about that, or if you'd still prefer changing everything to add those conditional checks. I typically prefer Interfaces and binding it from a service provider, but I'm definitely open to trying a different approach.

Reply:
> Let me know what you think about that, or if you'd still prefer changing everything to add those conditional checks. I typically prefer Interfaces and binding it from a service provider, but I'm definitely open to trying a different approach.

In a project context I like this too, but in a package context I tend to avoid interfaces unless it's a really good fit. 

Let's do the conditional check!

Reply:
Hey @freekmurze I went ahead and implemented a new solution where we now have a method within the service provider that checks to make sure the custom model extends our package model :) 

I think this might be a much cleaner way of implementing the check compared to adding conditionals everywhere we need the models. 

What do you think?

Reply:
Thanks for your work. The implementation can be improved imho, but you could already write some tests around it so things can be refactored safely.

Reply:
@freekmurze I'll work on getting the tests written 👍